Indoor_GPS/GPS.py

```diff
@@ -0,0 +1,62 @@
+from Indoor_GPS.marvelmind import MarvelmindHedge
+from Constants.Consts import *
+import time
+
+class GPS():
+
+    def __init__(self, grid, pid):
+        # create Marvel Mind Hedge thread
+        # get USB port with ls /dev/tty.usb*
+        # adr is the address of the hedgehog beacon!
+        self.hedge = MarvelmindHedge(tty=tty, adr=adr, debug=False)
+        # start thread
+        self.hedge.start()
+        # REQUIRED SLEEP TIME in order for thread to start and init_pos to be correct
+        time.sleep(1)
+        # data in array's [usnadr, x, y, z, timestamp]
+        self.init_pos = self.hedge.position()
+        self.init_pos = [self.init_pos[0], self.init_pos[2], self.init_pos[1]*-1, self.init_pos[3], self.init_pos[4]]
+        self.location_buffer = []
+        self.grid = grid
+        self.pid = pid
+
+    def update_loc(self, curr_tile):
+        """
+        updates the current tile based on the GPS input
+        """
+        # call indoor gps get location function
+        avgPosition = [0, 0]
+        for prev_pos in self.location_buffer:
+            if prev_pos is not None:
+                avgPosition[0] += prev_pos[0]
+                avgPosition[1] += prev_pos[1]
+
+        if len(self.location_buffer) >= 5:
+            avgPosition[0] /= len(self.location_buffer)
+            avgPosition[1] /= len(self.location_buffer)
+            [_, y, x, z, ang, time] = self.hedge.position()
+            x = -x
+            x1 = x
+            y1 = y
+    
+            self.location_buffer.pop(0)
+            self.location_buffer.append((x1, y1))
+            # map the position to the correct frame of reference
+            x = (x - self.init_pos[1]) * 10
+            y = (y - self.init_pos[2]) * 10
```

Comment:
Why do they need to be multiplied by 10?

Reply:
This, and lines 38-40, is a scale/reflect transform that we did in order to get the coordinates to be oriented/scaled nicely with the dimensions of the lab. I guess in the future we will have to do some experiments/measurements to make sure that we can convert these to meters for conveniently parsing path planning input commands.